lib/spack/spack/fetch_strategy.py

```diff
@@ -1388,6 +1359,45 @@ def fetch(self):
             raise FailedDownloadError(self.url)
 
 
+@fetcher
+class FetchAndVerifyExpandedFile(URLFetchStrategy):
+    """Fetch strategy that verifies the content digest during fetching,
+    as well as after expanding it."""
+
+    def __init__(self, url, archive_sha256: str, expanded_sha256: str):
```

Comment:
fwiw, i wanted to do composition > inheritance, but I think some code paths do isinstance(URLFetchStrategy), so not very helpful.